src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -875,8 +848,12 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
             current_hidden_states = expert_layer(current_state) * routing_weights[top_x, idx, None]
 
             # However `index_add_` only support torch tensors for indexing so we'll use
-            # the `top_x` tensor here.
-            final_hidden_states.index_add_(0, top_x, current_hidden_states.to(hidden_states.dtype))
+            # the `top_x` tensor here. this will give `skipping cudagraphs due to index put with accumulate`
+            # in compile
+            # final_hidden_states.index_add_(0, top_x, current_hidden_states.to(hidden_states.dtype))
+
+            # still suffers from `skipping cudagraphs due to ['incompatible ops']`
+            final_hidden_states[top_x] += current_hidden_states.to(hidden_states.dtype)
```

Comment:
Here seems to give `cudagraph warnings` no matter what equivalent form I put

Reply:
I think this is expected pretty much yes! 
If we use the megablock like implementation (with sparse topology and matrix reprensentation) like it was done in JetMoE we might be able to get over this, but not sure we can go further with the current version!

Reply:
> I think this is expected pretty much yes! If we use the megablock like implementation (with sparse topology and matrix reprensentation) like it was done in JetMoE we might be able to get over this, but not sure we can go further with the current version!

Yes, the root cause is `top_x`  here we use is `unbacked free symbols` in torch.compile and is data dependent beucase of `torch.where`,  this will cause skipped cudagraphs, but we will still benefit from partial cudagraphs if we are not rewriting it into sparse forms

Reply:
unfortunately, currently `torch.compile` produces wrong results, even when setting `fullgraph=False` and do not use `cudagraphs`, I believe it has something to do with `torch.where` used here, the traced fx graph is not correct, I think if we want to support `torch.compile` we have to rewrite it in another way